src/test/obj_ulog_size/obj_ulog_size.c

```diff
@@ -130,27 +154,302 @@ do_tx_max_alloc_tx_publish_abort(PMEMobjpool *pop)
 		UT_FATAL("Can publish");
 	} TX_END
 
-	free_pool(oid);
+	free_pool(oids, noids);
 	pmemobj_cancel(pop, act, REDO_OVERFLOW);
 }
 
+/*
+ * do_clt_overhead_test -- ctls read overhead and alignment
+ * for buffer in ulog
+ */
+static void
+do_clt_overhead_test(PMEMobjpool *pop)
+{
+	int ret;
+	unsigned ro_value = 0;
+
+	ret = pmemobj_ctl_get(pop, "tx.snapshot.alignment", &ro_value);
+	UT_ASSERTeq(ret, 0);
+	UT_ASSERTeq(ro_value, TX_SNAPSHOT_ALIGNMENT);
+
+	ret = pmemobj_ctl_get(pop, "tx.intent.alignment", &ro_value);
+	UT_ASSERTeq(ret, 0);
+	UT_ASSERTeq(ro_value, TX_INTENT_ALIGNMENT);
+
+	ret = pmemobj_ctl_get(pop, "tx.snapshot.log.entry_overhead", &ro_value);
+	UT_ASSERTeq(ret, 0);
+	UT_ASSERTeq(ro_value, TX_SNAPSHOT_LOG_ENTRY_OVERHEAD);
+
+	ret = pmemobj_ctl_get(pop, "tx.intent.log.entry_overhead", &ro_value);
+	UT_ASSERTeq(ret, 0);
+	UT_ASSERTeq(ro_value, TX_INTENT_LOG_ENTRY_OVERHEAD);
+
+	ret = pmemobj_ctl_get(pop, "tx.snapshot.log.buffer_overhead",
+			&ro_value);
+	UT_ASSERTeq(ret, 0);
+	UT_ASSERTeq(ro_value, TX_SNAPSHOT_LOG_BUFFER_OVERHEAD);
+
+	ret = pmemobj_ctl_get(pop, "tx.intent.log.buffer_overhead", &ro_value);
+	UT_ASSERTeq(ret, 0);
+	UT_ASSERTeq(ro_value, TX_INTENT_LOG_BUFFER_OVERHEAD);
+}
+
+/*
+ * do_tx_max_alloc_no_user_alloc_snap -- fills the pool and tries to do
+ * snapshot which is bigger than ulog size
+ */
+static void
+do_tx_max_alloc_no_user_alloc_snap(PMEMobjpool *pop)
+{
+	UT_OUT("do_tx_max_alloc_no_user_alloc_snap");
+	size_t noids = 0;
+	PMEMoid *oids = fill_pool(pop, &noids);
+	UT_ASSERT(noids >= MIN_NOIDS);
+
+/*
+ * querying snapshot size
+ * oids[0] object is taken intentionally - size of the pool is 16 MB,
+ * allocation starts with objects with size of 1 MB, that's mean that
+ * oids[0] is bigger than default undo log, which size is 2 KB
+ */
+	size_t snap_size =
+		pmemobj_alloc_usable_size(oids[LOG_BUFFER]);
+	TX_BEGIN(pop) {
+		/* it should abort - cannot allocate memory */
+		pmemobj_tx_add_range(oids[RANGE], 0, snap_size);
+	} TX_ONABORT {
+		UT_OUT("!Cannot add snapshot");
+	} TX_ONCOMMIT {
+		UT_FATAL("Can add snapshot");
+	} TX_END
+
+	free_pool(oids, noids);
+}
+
+/*
+ * do_tx_max_alloc_user_alloc_snap -- fills the pool, apends allocated
+ * buffer and tries to do snapshot which is bigger than ulog size
+ */
+static void
+do_tx_max_alloc_user_alloc_snap(PMEMobjpool *pop)
+{
+	UT_OUT("do_tx_max_alloc_user_alloc_snap");
+	size_t noids = 0;
+	PMEMoid *oids = fill_pool(pop, &noids);
+	UT_ASSERT(noids >= MIN_NOIDS);
+
+	/* querying snapshot size */
+	size_t buff_size =
+		pmemobj_alloc_usable_size(oids[LOG_BUFFER]);
+	void *buff_addr = pmemobj_direct(oids[LOG_BUFFER]);
+	size_t snap_size =
+		pmemobj_alloc_usable_size(oids[LOG_BUFFER + 1]);
+
+	TX_BEGIN(pop) {
+		pmemobj_tx_log_append_buffer(
+			TX_LOG_TYPE_SNAPSHOT, buff_addr, buff_size);
+		pmemobj_tx_add_range(oids[RANGE], 0, snap_size);
```

Comment:
I do not understand this, why snap size is from different oid than oid passed to pmemobj_tx_add_range. Or it is the same?
\+ @janekmi